Monotype/c2patool
# C2PA-484: Pull in `main`, update `c2pa-rs` to most recent `monotype/fontSupport` branch

## Changes in this pull request

This PR first updates the `main` branch of `c2patool` and merges it into our `monotype/fontSupport` branch (fixing up merge conflicts in the process).  Then it updates the `c2pa` dependency to get the most recent `c2pa-rs` `monotype.fontSupport` branch to pick up support for DSIG dummy table generation.

## Verification

- Build the `cp2atool`
  - `cargo build`
- Sign a font with it
  - `.\target\debug\c2patool.exe -m .\sample\test.json "CultStd_original.otf" -f -d -o "CultStd_original_signed.otf"`
- Inspect the font (using `ttx` or similar tool) to verify that the `DSIG` table is empty
- Inspect the font with `c2patool` to verify that it was correctly signed and that all signature/hashes are valid
   - `.\target\debug\c2patool.exe -d "CultStd_original_signed.otf"`

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.
